grpc/service_config/service_config.proto

```diff
@@ -215,47 +236,94 @@ message WeightedTargetLoadBalancingPolicyConfig {
-  // the server name from the target URI.
-  string eds_service_name = 4;
-  // LRS server to send load reports to.
-  // If not present, load reporting will be disabled.
-  // If set to the empty string, load reporting will be sent to the same
-  // server that we obtained CDS data from.
-  google.protobuf.StringValue lrs_load_reporting_server_name = 5;
+// Configuration for xds_cluster_resolver LB policy.
+message XdsClusterResolverLoadBalancingPolicyConfig {
+  // Describes a discovery mechanism instance.
+  // For EDS or LOGICAL_DNS clusters, there will be exactly one
+  // DiscoveryMechanism, which will describe the cluster of the parent
+  // CDS policy.
+  // For aggregate clusters, there will be one DiscoveryMechanism for each
+  // underlying cluster.
+  message DiscoveryMechanism {
+    // Cluster name.
+    string cluster = 1;
+
+    // LRS server to send load reports to.
+    // If not present, load reporting will be disabled.
+    // If set to the empty string, load reporting will be sent to the same
+    // server that we obtained CDS data from.
+    google.protobuf.StringValue lrs_load_reporting_server_name = 2;
+
+    // Maximum number of outstanding requests can be made to the upstream
+    // cluster.  Default is 1024.
+    google.protobuf.UInt32Value max_concurrent_requests = 3;
+
+    enum Type {
+      UNKNOWN = 0;
+      EDS = 1;
+      LOGICAL_DNS = 2;
+    };
+    Type type = 4;
+
+    // For type EDS only.
+    // EDS service name, as returned in CDS.
+    // May be unset if not specified in CDS.
+    string eds_service_name = 5;
+
+    // For type LOGICAL_DNS only.
+    // DNS name to resolve in "host:port" form.
+    string dns_hostname = 6;
+  }
+
+  // Ordered list of discovery mechanisms.
+  // Must have at least one element.
+  // Results from each discovery mechanism are concatenated together in
+  // successive priorities.
+  repeated DiscoveryMechanism discovery_mechanisms = 1;
+
+  // xDS LB policy.
+  // This represents the xDS LB policy, which does not necessarily map
+  // one-to-one to a gRPC LB policy.  Currently, the following policies
+  // are supported:
+  // - "ROUND_ROBIN" (config is empty)
+  // - "RING_HASH" (config is a RingHashLoadBalancingConfig)
+  repeated LoadBalancingConfig xds_lb_policy = 2;
+}
+
+// Configuration for xds_cluster_impl LB policy.
+message XdsClusterImplLoadBalancingPolicyConfig {
+  // Cluster name.  Required.
+  string cluster = 1;
+
+  // EDS service name.
+  // Not set if cluster is not an EDS cluster or if it does not
+  // specify an EDS service name.
+  string eds_service_name = 2;
+
+  // Server to send load reports to.
+  // If unset, no load reporting is done.
+  // If set to empty string, load reporting will be sent to the same
+  // server as we are getting xds data from.
+  google.protobuf.StringValue lrs_load_reporting_server_name = 3;
+
+  // Maximum number of outstanding requests can be made to the upstream cluster.
+  // Default is 1024.
```

Comment:
Is "0" valid here?

Reply:
Yes.  It means 0 concurrent requests are allowed.